src/samples/audio/smart_amp_test.c

```diff
@@ -86,6 +105,200 @@ static struct comp_dev *smart_amp_new(const struct comp_driver *drv,
+		sink_c = buffer_acquire(sink);
+		sink_c->stream.channels = out_fmt.channels_count;
+		sink_c->stream.rate = out_fmt.sampling_frequency;
+
+		audio_stream_fmt_conversion(out_fmt.depth,
+					    out_fmt.valid_bit_depth,
+					    &sink_c->stream.frame_fmt,
+					    &sink_c->stream.valid_sample_fmt,
+					    out_fmt.s_type);
+
+		sink_c->buffer_fmt = out_fmt.interleaving_style;
+		params->frame_fmt = sink_c->stream.frame_fmt;
+
+		sink_c->hw_params_configured = true;
+
+		buffer_release(sink_c);
+	}
+}
+
+static inline int smart_amp_set_config(struct comp_dev *dev, const char *data,
+				       uint32_t data_size)
+{
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+	struct sof_smart_amp_config *cfg;
+	uint32_t cfg_size;
+
+	cfg = (struct sof_smart_amp_config *)data;
+	cfg_size = data_size;
+
+	if (cfg_size != sizeof(struct sof_smart_amp_config)) {
+		comp_err(dev, "smart_amp_set_config(): invalid config size %u, expect %u",
+			 cfg_size, sizeof(struct sof_smart_amp_config));
+		return -EINVAL;
+	}
+
+	comp_dbg(dev, "smart_amp_set_config(): config size = %u", cfg_size);
+
+	memcpy_s(&sad->config, sizeof(struct sof_smart_amp_config), cfg,
+		 sizeof(struct sof_smart_amp_config));
+
+	return 0;
+}
+
+static inline int smart_amp_get_config(struct comp_dev *dev, char *data,
+				       uint32_t *data_size)
+{
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+	uint32_t cfg_size;
+
+	cfg_size = sizeof(struct smart_amp_data);
+
+	if (cfg_size > *data_size) {
+		comp_err(dev, "smart_amp_get_config(): wrong config size %d",
+			 *data_size);
+	}
+
+	*data_size = cfg_size;
+
+	return memcpy_s(data, cfg_size, &sad->config, cfg_size);
+}
+
+static int smart_amp_set_large_config(struct comp_dev *dev, uint32_t param_id, bool first_block,
+				      bool last_block, uint32_t data_offset, const char *data)
+{
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+
+	comp_dbg(dev, "smart_amp_set_large_config()");
+
+	switch (param_id) {
+	case SMART_AMP_SET_MODEL:
+		return ipc4_comp_data_blob_set(sad->model_handler,
+					       first_block,
+					       last_block,
+					       data_offset,
+					       data);
+	case SMART_AMP_SET_CONFIG:
+		return smart_amp_set_config(dev, data, data_offset);
+	default:
+		return -EINVAL;
+	}
+}
+
+static int smart_amp_get_large_config(struct comp_dev *dev, uint32_t param_id, bool first_block,
+				      bool last_block, uint32_t *data_offset, char *data)
+{
+	comp_dbg(dev, "smart_amp_get_large_config()");
+
+	switch (param_id) {
+	case SMART_AMP_GET_CONFIG:
+		return smart_amp_get_config(dev, data, data_offset);
+	default:
+		return -EINVAL;
+	}
+}
+
+static int smart_amp_get_attribute(struct comp_dev *dev, uint32_t type,
+				   void *value)
+{
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+
+	comp_dbg(dev, "smart_amp_get_attribute()");
+
+	switch (type) {
+	case COMP_ATTR_BASE_CONFIG:
+		*(struct ipc4_base_module_cfg *)value = sad->config.base;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
```

Comment:
nitpick: if you do a revision, you could move this to line 238 instead of `break`

Reply:
sure, thanks